lib/rucio/common/utils.py

```diff
@@ -1342,6 +1343,15 @@ def func(*args):
     return logger
 
 
+def daemon_sleep(start_time, sleep_time, graceful_stop):
```

Comment:
It would be good to foresee the handthrough of the logger object there, so we can also do logging with the thread prefixes for the sleep log-message. (Have a look at other methods which pass through the logger object there)
Otherwise it looks good 👍